src/tools/chdman.cpp

```diff
@@ -1039,6 +1117,18 @@ static void parse_input_chd_parameters(const parameters_map &params, chd_file &i
 		chd_error err = input_chd.open(input_chd_str->second->c_str(), writeable, input_parent_chd.opened() ? &input_parent_chd : nullptr);
 		if (err != CHDERR_NONE)
 			report_error(1, "Error opening CHD file (%s): %s", input_chd_str->second->c_str(), chd_file::error_string(err));
+		if (input_chd_parent_str == params.end() && input_chd.parent_sha1() != util::sha1_t::null)
+    {
+			std::string chd_dir;
+			const size_t sep_index = input_chd_str->second->rfind(PATH_SEPARATOR);
```

Comment:
You need to use `util::is_directory_separator`, you can’t assume there’s only one path separator character.

Reply:
Actually, we don't care how many separators there are. This is just getting the directory the CHD file is in, that is, the string without the filename, which is just after the LAST separator. The only corner case is if there are 0 separators, which we check for.

Reply:
Preprocessor? Why not just use `std::find_if`?  See here for one example: https://github.com/mamedev/mame/blob/mame0226/src/emu/render.cpp#L2176

Reply:
Done